packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -123,27 +235,39 @@ class ModalBarrier extends StatelessWidget {
       }
     }
 
+    Widget barrier = Semantics(
+      onTapHint: semanticsOnTapHint,
+      onTap: handleDismiss,
+      label: semanticsDismissible ? semanticsLabel : null,
+      textDirection: semanticsDismissible && semanticsLabel != null ? Directionality.of(context) : null,
+      child: MouseRegion(
+        cursor: SystemMouseCursors.basic,
+        child: ConstrainedBox(
+        constraints: const BoxConstraints.expand(),
+        child: color == null ? null : ColoredBox(
+          color: color!,
+          ),
+        ),
+      ),
+    );
+
+    final bool excluding = !semanticsDismissible || !modalBarrierSemanticsDismissible;
+
+    if (!excluding && clipDetailsNotifier != null) {
+      barrier = _SemanticsClipper(
+        clipDetailsNotifier: clipDetailsNotifier,
+        child: barrier,
+      );
+    }
```

Comment:
Is this necessary? If there's no impact, consider always using `_SemanticsClipper` and moving back `barrier` to line 270 below

Reply:
Thanks! Moved back barrier